chain-impl-mockchain/src/leadership/genesis/mod.rs

```diff
@@ -295,6 +295,49 @@ mod tests {
         );
     }
 
+    #[test]
+    pub fn test_leader_with_invalid_pool_id() {
+        // Arrange
+        let leader_election_parameters = LeaderElectionParameters::new();
+
+        let cb = ConfigBuilder::new()
+            .with_slots_per_epoch(leader_election_parameters.slots_per_epoch)
+            .with_active_slots_coeff(leader_election_parameters.active_slots_coeff_as_milli());
+
+        let mut ledger = LedgerBuilder::from_config(cb)
+            .build()
+            .expect("cannot build test ledger")
+            .ledger;
+
+        let mut pools = HashMap::<PoolId, (SecretKey<RistrettoGroup2HashDh>, u64, Stake)>::new();
+
+        for _i in 0..leader_election_parameters.pools_count {
```

Comment:
nitpick:

this can be written as generator:

```
let mut pools: HashMap<_>  = std::iter::from_fn(|| {
      let (pool_id, pool_vrf_private_key) = make_pool(&mut ledger);
      (
                pool_id.clone(),
                (pool_vrf_private_key, 0, leader_election_parameters.value),
            )
 }).take(leader_election_parameters.pools_count).collect();
```

and you dont need line 312 i guess